mediaproviders/src/main/java/com/kaltura/playkit/providers/ovp/KalturaOvpMediaProvider.java

```diff
@@ -361,13 +367,40 @@ public static PKMediaEntry getMediaEntry(String baseUrl, String ks, String partn
             populateMetadata(metadata, entry);
             PKMediaEntry mediaEntry = initPKMediaEntry(entry.getTags());
 
+            if (useApiCaptions && playbackContext.getPlaybackCaptions() != null && !playbackContext.getPlaybackCaptions().isEmpty()) {
+                mediaEntry.setExternalSubtitleList(createExternalSubtitles(playbackContext));
+            }
+
             return mediaEntry.setId(entry.getId()).setSources(sources)
                     .setDuration(entry.getMsDuration())
                     .setMetadata(metadata)
                     .setName(entry.getName())
                     .setMediaType(MediaTypeConverter.toMediaEntryType(entry));
         }
 
+        private static List<PKExternalSubtitle> createExternalSubtitles(KalturaPlaybackContext playbackContext) {
+            List<PKExternalSubtitle> subtitleList = new ArrayList<>();
+            List<KalturaPlaybackCaption> playbackCaptionList = playbackContext.getPlaybackCaptions();
+
+            for (KalturaPlaybackCaption kalturaPlaybackCaption : playbackCaptionList) {
+
+                String subtitleURL = kalturaPlaybackCaption.getWebVttUrl();
+
+                PKExternalSubtitle pkExternalSubtitle = new PKExternalSubtitle()
+                        .setUrl(subtitleURL)
+                        .setMimeType(PKSubtitleFormat.valueOfUrl(subtitleURL))
+                        .setLabel(kalturaPlaybackCaption.getLabel())
+                        .setLanguage(kalturaPlaybackCaption.getLanguage());
```

Comment:
use `languageCode` instead - this is the field which is used to populate the manifests as well

Reply:
done